numba_cuda/numba/cuda/cuda_paths.py

```diff
@@ -251,120 +284,196 @@ def get_system_ctk(*subdirs):
 
-    versions = (12, 13)
-    for version in versions:
-        if version == 12:
-            cuda_include_path = os.path.join(
-                nvidia_path, "cuda_runtime", "include"
-            )
-        elif version == 13:
-            cuda_include_path = os.path.join(nvidia_path, "cu13", "include")
-        else:
-            raise ValueError(f"Unsupported version: {version}")
-
-        if os.path.exists(os.path.join(cuda_include_path, "cuda.h")):
-            return cuda_include_path
+def get_conda_ctk_libdir():
+    """Return path to directory containing the shared libraries of cudatoolkit."""
+    is_conda_env = os.path.isdir(os.path.join(sys.prefix, "conda-meta"))
+    if not is_conda_env:
+        return None
+    libdir = os.path.join(
+        sys.prefix,
+        "Library" if IS_WIN32 else "lib",
+        "bin" if IS_WIN32 else "",
+    )
+    # Windows CUDA 13.0.0 uses "bin\x64" directory but 13.0.1+ just uses "bin" directory
+    if IS_WIN32 and os.path.isdir(os.path.join(libdir, "x64")):
+        libdir = os.path.join(libdir, "x64")
+    # Assume the existence of nvrtc to imply needed CTK libraries are installed
+    paths = find_lib("nvrtc", libdir)
+    if not paths:
+        return None
+    # Use the directory name of the max path
+    return os.path.dirname(max(paths))
 
 
-def get_nvidia_nvvm_ctk():
-    """Return path to directory containing the NVVM shared library."""
-    is_conda_env = os.path.exists(os.path.join(sys.prefix, "conda-meta"))
+def get_libdevice_conda_path():
+    """Return path to directory containing the libdevice bitcode library."""
+    is_conda_env = os.path.isdir(os.path.join(sys.prefix, "conda-meta"))
     if not is_conda_env:
-        return
+        return None
 
-    # Assume the existence of NVVM in the conda env implies that a CUDA toolkit
-    # conda package is installed.
-    if IS_WIN32:
-        # The path used on Windows
-        libdir = os.path.join(sys.prefix, "Library", "nvvm", _cudalib_path())
-    else:
-        # The path used on Linux is different to that on Windows
-        libdir = os.path.join(sys.prefix, "nvvm", _cudalib_path())
+    # Linux: nvvm/libdevice/libdevice.10.bc
+    # Windows: Library/nvvm/libdevice/libdevice.10.bc
+    libdevice_path = os.path.join(
+        sys.prefix,
+        "Library" if IS_WIN32 else "",
+        "nvvm",
+        "libdevice",
+        "libdevice.10.bc",
+    )
+    if os.path.isfile(libdevice_path):
+        return libdevice_path
+    return None
 
-    if not os.path.exists(libdir) or not os.path.isdir(libdir):
-        # If the path doesn't exist, we didn't find the NVIDIA conda package
-        return
 
-    paths = find_lib("nvvm", libdir=libdir)
-    if not paths:
-        return
-    # Use the directory name of the max path
-    return os.path.dirname(max(paths))
+def _get_nvvm_conda_path():
+    """Return path to directory containing the nvvm library."""
+    is_conda_env = os.path.isdir(os.path.join(sys.prefix, "conda-meta"))
```

Comment:
Does this require the env to be activated?

Reply:
I was initially worried this might grab an nvvm from an unactivated environment but I don't think it's a concern.